_posts/2020-06-10-Solidity-0610-release-announcement.md

```diff
@@ -0,0 +1,113 @@
+---
+layout: post
+published: true
+title: 'Solidity 0.6.10 Release Announcement'
+date: '2020-06-11'
+author: Solidity Team
+category: Releases
+---
+
+[Solidity v0.6.10](https://github.com/ethereum/solidity/releases/tag/v0.6.10) fixes an important bug that was introduced in the previous release and adds error codes.
+
+Please upgrade if you are using internal library functions with calldata paramaters in connection with ``using for``.
+
+## Important Bugfixes
+
+### Invalid Access for Calldata Parameters in Internal Library Functions
+
+**Problem**: The previous release Solidity 0.6.9 introduced the possibility to use ``calldata`` types in internal and not only in external functions.
+This resulted in a bug related to calling library functions with calldata parameters via ``using for``.
+More specifically, the compiler would copy all calldata arguments to memory first - like is done for calls to public library functions.
+The internal library function would then receive the memory pointer but interpret it as a calldata pointer resulting in
+calldata reads at the wrong location.
+
+We assigned a severity of "very low" due to the fact that the bug was only introduced a week ago
+and is very easy to detect through testing.
+
+**Solution**: Upgrade to Solidity 0.6.10 that does not perform the copy to memory.
+
+**Example of affected code**
+
+```solidity
+library L {
+    function helper(bytes calldata _x) internal pure returns (byte) {
+        return _x[0];
+    }
+}
+
+contract C {
+    using L for bytes;
+    function fun(bytes calldata _x) public pure returns (byte) {
+        // This would incorrectly copy ``_x`` from calldata
+        // to memory and then call L.helper with this memory array,
+        // but the function expects a calldata array and thus
+        // reads the data from the wrong calldata offset.
+        return _x.helper();
+    }
+}
+```
+
+We especially thank Luis Schliesske ([@gitpusha](https://github.com/gitpusha)) for reporting the bug!
+
+
+### New Feature: Error codes
+
+A notable new feature is the introduction of error codes. On the commandline, you can enable
+error codes via the switch ``--error-codes`` while via standard-json, the error codes will
+always be available under the property ``"errorCode"``.
```

Comment:
I am not sure if "while via standard-json" is grammatically correct. Maybe:
In Standard JSON input/output mode (`--standard-json`) the error codes will always be available under the property ``"errorCode"``. In the command-line interface, you can enable error codes via the switch ``--error-codes``.

**Disclaimer** I am not a native speaker. Not at all :).

Reply:
I think it's fine how it is. At least I understand it. I'm not 100% sure on the `,`after json but it should be fine.